### PR TITLE
Add a ValidationLieGroup

### PR DESCRIPTION
**Pull request by kellertuer**
📋

This PR aims to provide the analogue of a [`ValidationManifold`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/manifolds/#A-manifold-for-validation) for Lie groups. Internally one can (but does not have to) add such a validation also for the inner manifold functions.

* [x] implement the Lie groups interface
* [x] 📚 adapt docs
* [x] resolve ambiguties 
* [x] test coverage

### CODE REVIEW

**Comment by kellertuer** [comment]
This basically just needs a few functions to be defined for the `LieAlgebra` of a `ValidatinLieGroup` to handle wrapped points a bit – and a careful check, whether or not `Identity(G)` should be wrapped as `ValidationMPoint(Identity(G))` or not.

**Comment by kellertuer** [comment]
Decided again st wrapping identity, since that causes too much strange side effects on the system.

**Comment by kellertuer** [comment]
I consider this ready for review, since the only thing that should be missing by now is a few test coverage corner cases.

**Comment by kellertuer** [comment]
The only lines missing are the ones that validate `jacobian_conjugate`, which we can do a test for as soon as we have the default from #20.

So this PR is complete.

File: src/groups/validation_group.jl [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+* `:All`: disable all checks
+* `:Point`: checks for points
+* `:Algebra`: checks related to the [`LieAlgebra`](@ref)
+* `:Output`: checks for output
+* `:Input`: checks for input variables
+
+# Fields
+
+* `lie_group::L` the [`AbstractLieGroup`](@ref) to be decorated
+* ``
```
Comment (mateuszbaran) [inline src/groups/validation_group.jl]: Other fields seem to be missing, though we could probably just link to `ValidationManifold` for their description`?

File: src/groups/validation_group.jl [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+    error=G.mode,
+    within::Union{Nothing,<:Function}=nothing,
+    context::Union{NTuple{N,Symbol} where N}=NTuple{0,Symbol}(),
+)
+    !_vLc(G, within, context) && return nothing
+    (error === :error) && (throw(ErrorException(str)))
+    (error === :warn) && (@warn str)
+    (error === :info) && (@info str)
+    return nothing
+end
```
Comment (mateuszbaran) [inline src/groups/validation_group.jl:167]: Could you maybe reuse `_msg` from ManifoldsBase? If you extend `_vMc` instead of defining `_vLc` it should work fine I think?

File: src/groups/validation_group.jl (hunk 1 above)
Comment (kellertuer) [reply src/groups/validation_group.jl]: Ah I forgot to add them, yes it is basically the same fields (again) as for the Validation one, and I would even try to keep both parallel – since that way one can pass down the same things and validate on M as well – or not – if one wants to.

**Comment by mateuszbaran** [comment]
I have only these two comments, everything else looks fine.

File: src/groups/validation_group.jl (hunk 2 above)
Comment (kellertuer) [reply src/groups/validation_group.jl:167]: there was some reason why I had to rewrite `_vMc` to `_vLc`, the `_msg` I then rewrote to not depend on ManifoldsBase internals – though that is of course more of a philosophical point – that way we can change the internals of ManifoldsBase without breaking stuff here.

File: src/groups/validation_group.jl (hunk 1 above)
Comment (kellertuer) [reply src/groups/validation_group.jl]: Fixed it :)

**Comment by kellertuer** [comment]
It was also mainly a challenge to work through all the functions and not necessarily challenging from the intellectual side ;)

File: src/groups/validation_group.jl (hunk 2 above)
Comment (mateuszbaran) [reply src/groups/validation_group.jl:167]: I see. I would prefer less duplication but we can keep it this way too.

**Review by mateuszbaran — APPROVED** [review]

File: src/groups/validation_group.jl (hunk 2 above)
Comment (kellertuer) [reply src/groups/validation_group.jl:167]: I do see positive points in both approaches. Mine is a bit more independent of internal things of another package – yours avoids a few lines of code duplication. But thanks that we can leave it like this for now.

If we come across a good reason to have these few lines not double, we can also change it later.

**Comment by kellertuer** [comment]
Nice! Then let's merge this one first, I can care about the missing code coverage in the jacobian PR then.
